draginowqslb/sensor.go

```diff
@@ -0,0 +1,243 @@
+) (sensor.Sensor, error) {
+	n := &WQSLB{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+		node:   node.NewSensor(conf, logger),
+	}
+
+	err := n.Reconfigure(ctx, deps, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return n, nil
+}
+
+// Reconfigure reconfigures the sensor.
+func (n *WQSLB) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	cfg, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	nodeCfg := cfg.getNodeConfig()
+
+	if err = n.node.ReconfigureWithConfig(ctx, deps, &nodeCfg); err != nil {
+		return err
+	}
+
+	// call this once outside of background thread to get any info gateway has before calling the interval request
+	n.node.GetAndUpdateDeviceInfo(ctx)
+
+	// Set the interval if one was provided
+	// We do not send a default in case the user has already set an interval they prefer
+	if cfg.Interval != nil && *cfg.Interval != 0 {
+		if _, err := dragino.SendIntervalDownlink(ctx, &n.node, *nodeCfg.Interval, false); err != nil {
+			return err
+		}
+	}
+
+	if err = node.CheckCaptureFrequency(conf, *nodeCfg.Interval, n.logger); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Close removes the device from the gateway.
+func (n *WQSLB) Close(ctx context.Context) error {
+	return n.node.Close(ctx)
+}
+
+// Readings returns the node's readings.
+func (n *WQSLB) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
+	return n.node.Readings(ctx, extra)
+}
+
+// DoCommand implements the DoCommand interface.
+func (n *WQSLB) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+	testOnly := node.CheckTestKey(cmd)
+
+	if interval, intervalSet := cmd[node.IntervalKey]; intervalSet {
+		if intervalFloat, floatOk := interval.(float64); floatOk {
+			return dragino.SendIntervalDownlink(ctx, &n.node, intervalFloat, testOnly)
+		}
+		return map[string]interface{}{}, fmt.Errorf("error parsing payload, expected float got %v", reflect.TypeOf(interval))
+	}
+	if _, ok := cmd[node.ResetKey]; ok {
+		return dragino.SendResetDownlink(ctx, &n.node, testOnly)
+	}
+	// WQSLB specific calibration commands
+	if ph, ok := cmd["calibrate_ph"]; ok {
+		phFloat, ok := ph.(float64)
+		if !ok {
+			return map[string]interface{}{}, fmt.Errorf("expected float64 got %v", reflect.TypeOf(ph))
+		}
+		payload := "FB"
+		switch phFloat {
+		case 9:
+			payload += "09"
+		case 6:
+			payload += "06"
+		case 4:
+			payload += "04"
+		default:
+			return map[string]interface{}{}, fmt.Errorf("unexpected ph calibration value %f, valid values are 4, 6, or 9", phFloat)
+		}
+		return n.node.SendDownlink(ctx, payload, testOnly)
+	}
+
+	if ec, ok := cmd["calibrate_ec"]; ok {
+		ecFloat, ok := ec.(float64)
+		if !ok {
+			return map[string]interface{}{}, fmt.Errorf("expected float64 got %v", reflect.TypeOf(ec))
+		}
+		payload := "FD"
+		switch ecFloat {
+		case 1:
+			payload += "01"
+		case 10:
```

Comment:
Similarly here and for the rest of the calibration DoCommands - comment which step corresponds to which solution state.

We can be lazy this way and not look at the reamde